src/Google/Cache/File.php

```diff
@@ -155,7 +155,7 @@ private function getCacheDir($file, $forWrite)
     // trim the directory separator from the path to prevent double separators
     $storageDir = rtrim($this->path, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $dirHash;
     if ($forWrite && ! is_dir($storageDir)) {
-      if (! mkdir($storageDir, 0700, true)) {
+      if (!mkdir($storageDir, 0777, true)) {
```

Comment:
Access tokens are stored in cache, yes. But the whole point is to fix #806. So we either break some of our users and be extra secure, or choose to be standard with other file caches and make file permissions looser.